src/stories/containers/RecognizedDelegates/components/FilterDelegate/DelegateSelectItem.tsx

```diff
@@ -80,9 +80,21 @@ const CheckWrapper = styled.span({
 const StyledContainer = styled(Container)<{ isLight: boolean; checked: boolean; isDisable: boolean }>(
   ({ isDisable, checked, isLight }) => ({
     '&:hover': {
-      background: isDisable ? 'none' : isLight ? (checked ? '#EDEFFF' : '#F6F8F9') : isDisable ? 'none' : '#25273D',
-      backgroundColor:
-        isLight && checked ? '#EDEFFF' : isLight && !checked ? 'none' : !isLight && !checked ? '#000A13' : '#231536',
+      background: isLight
+        ? isDisable && checked
+          ? '#EDEFFF'
+          : isDisable && !checked
+          ? 'none'
+          : !isDisable && checked
+          ? '#EDEFFF'
+          : '#F6F8F9'
+        : isDisable && checked
+        ? '#231536'
+        : isDisable && !checked
+        ? 'none'
+        : !isDisable && checked
+        ? '#231536'
+        : '#25273D',
```

Comment:
As this has 3 different variables, it is pretty hard to follow, would be nice to create a variable in the styled component function to make it more readable. Maybe something like this:
```ts
const StyledContainer = styled(Container)<{ isLight: boolean; checked: boolean; isDisable: boolean }>(
  ({ isDisable, checked, isLight }) => {
  let background: string;
  if (isLight) {
    if (isDisabled) {
      background = checked ? 'xxx' : 'yyy';
    } else {
      background = checked ? 'qqq' : 'www';
    }
  } else {
    // ...
  }

  return ({
    '&:hover': {
      background,
    }
  })
});
```

Reply:
I got your point